packages/app-admin-users-cognito/src/plugins/userMenu/userInfo.tsx

```diff
@@ -62,70 +63,97 @@ const linkStyles = css({
+     */
+    accountRoute?: `/${string}`;
+}
+
+export const UserInfo = ({ accountRoute }: UserInfoProps) => {
     const security = useSecurity();
 
     if (!security || !security.identity) {
         return null;
     }
 
-    // This is only applicable in multi-tenant environments
-    const { currentTenant, defaultTenant } = security.identity;
+    const { profile, displayName } = security.identity;
 
-    let wrapper: any = { Component: Link, props: { to: "/account" } };
-    if (currentTenant && defaultTenant && currentTenant.id !== defaultTenant.id) {
-        wrapper = { Component: "div", props: {} };
-    }
+    // Start with the assumption that the user doesn't have a profile in the system (external IDP).
+    let listItem: JSX.Element = <UserInfoListItem displayName={displayName} />;
 
-    const profile = security.identity.profile;
-    if (!profile) {
-        const { displayName } = security.identity;
-
-        return (
-            <wrapper.Component {...wrapper.props} className={linkStyles}>
-                <ListItem ripple={false} className={linkStyles}>
-                    <ListItemGraphic className={avatarImage}>
-                        <Avatar
-                            className={"avatar"}
-                            src={undefined}
-                            alt={displayName}
-                            fallbackText={displayName}
-                            renderImage={props => <Image {...props} transform={{ width: 100 }} />}
-                        />
-                    </ListItemGraphic>
-                    <div>
-                        <h3>
-                            <Typography use={"headline6"}>{displayName}</Typography>
-                        </h3>
-                    </div>
-                </ListItem>
-            </wrapper.Component>
+    if (profile) {
+        const { email, firstName, lastName, avatar, gravatar } = profile;
+        const fullName = `${firstName} ${lastName}`;
+
+        listItem = (
+            <UserInfoListItem
+                displayName={fullName}
+                avatarSrc={avatar ? avatar.src : gravatar}
+                email={email}
+            />
         );
     }
 
-    const { email, firstName, lastName, avatar, gravatar } = profile;
-    const fullName = `${firstName} ${lastName}`;
+    return (
+        <UserInfoListItemContainer accountRoute={accountRoute} className={linkStyles}>
+            {listItem}
+        </UserInfoListItemContainer>
+    );
+};
+
+interface UserInfoListItemContainer {
+    accountRoute?: `/${string}`;
+    className: string;
+    children: React.ReactNode;
+}
+
+const UserInfoListItemContainer = ({
+    accountRoute,
+    className,
+    children
+}: UserInfoListItemContainer) => {
+    const isDefaultTenant = useIsDefaultTenant();
+
+    // If there's no "accountRoute", or the user is not on his default tenant, don't render the link to "Account Details".
```

Comment:
Hm is this true? I see in current system, we do show user info item when not on non-default tenant:

![image](https://github.com/webiny/webiny-js/assets/5121148/d35700e7-0c7e-480c-8c9e-6121f7857abb)

🤔 

The only thing is that when clicking on it, it doesn't lead anywhere.

Reply:
@adrians5j yes, we do show it, but it's not a link. It's a simple `div`.

Reply:
Ah srry my bad, I misunderstood the code.

All good now, ty!